include/ginkgo/core/base/batch_multi_vector.hpp

```diff
@@ -53,8 +53,10 @@ class MultiVector
     : public EnablePolymorphicObject<MultiVector<ValueType>>,
       public EnablePolymorphicAssignment<MultiVector<ValueType>>,
 #if GINKGO_ENABLE_HALF || GINKGO_ENABLE_BFLOAT16
-      public ConvertibleTo<
-          MultiVector<next_precision<next_precision<ValueType>>>>,
+      public ConvertibleTo<MultiVector<next_precision_move<ValueType, 2>>>,
+#endif
+#if GINKGO_ENABLE_HALF && GINKGO_ENABLE_BFLOAT16
+      public ConvertibleTo<MultiVector<next_precision_move<ValueType, 3>>>,
 #endif
```

Comment:
Should we perhaps always have these `ConvertibleTo` base classes? Right now, we always provide the `gko::half` and `gko::bfloat16` classes, so the `ConvertibleTo` variants are valid. It only requires making the `gko::array` fp16 conversion functions available, which is a very manageable cost compared to the full fp16 build.
From the user perspective, using (any) fp16 only breaks when they try to instantiate one of our classes. So they will not be able to use the conversion function with fp16 anyway.
I should note that the error will be a linker error, which is most likely very painful for the users to decipher.

Reply:
`ConvertibleTo` that class needs to instantiate the class then the member function will require the function all available.
We do not go the same way to deal with the dpcpp single (throw not_implment) because the current macro design to change that with the three conditions (single/half/bfloat) will be a disaster. also, I guess it will exceed the symbol limit in windows

Reply:
I might have not communicated exactly what I meant, so I will create a PR to illustrate. In short, I think the `#if` guards for the `ConvertibleTo` are unnecessary.